pkg/transporter/pipeline.go

```diff
@@ -118,16 +117,17 @@ func (pipeline *Pipeline) Run() error {
 		pipeline.Err = err // only set it if it hasn't been set already.
 	}
 
-	// pipeline has stopped, emit one last round of metrics and send the exit event
-	pipeline.emitMetrics()
 	if pipeline.sessionStore != nil {
 		pipeline.setState()
 	}
-	pipeline.source.pipe.Event <- events.NewExitEvent(time.Now().UnixNano(), VERSION, endpoints)
-
 	// the source has exited, stop all the other nodes
 	pipeline.Stop()
 
+	// pipeline has stopped, emit one last round of metrics and send the exit event
+	pipeline.emitMetrics()
+	pipeline.source.pipe.Event <- events.NewExitEvent(time.Now().UnixNano(), VERSION, endpoints)
+	pipeline.emitter.Stop()
+
```

Comment:
while making the changes to logging, I was seeing where the last emit'd metrics were almost never right for a sink node (typically 1 record less than the source), since we were emitting and then calling `Stop`, it introduced a period of time where the record counter hadn't been updated yet. so this change let's the pipeline fully shutdown before we emit any final events.